### ISSUE TITLE
Implement async test functions and inventory 

### ISSUE THREAD

**Issue by gusmb** [arista-netdevops-community/anta#43]
One of the main concerns for usability of the framework is performance and the fact that tests are run in sequence for each inventory device. Most of the tests are RPC calls over the network and creating connections that cause delays for am inventory of hundreds of devices. Would be nice to have an async method where all the tests (or just a subset) can run in parallel and waiting time could be used efficiently.
Also async inventory could initiate connection to all inventory hosts in parallel.

**Comment by mtache** [comment]
Hi !
PR #122 implement tests as coroutines and use an async library to create a device session in the InventoryDevice object.
This has been merged in the master branch.
Feel free to test and let us know if this issue can be closed.
